packages/core/elasticsearch/core-elasticsearch-client-server-internal/src/cluster_client.ts

```diff
@@ -146,4 +161,22 @@ export class ClusterClient implements ICustomClusterClient {
       ...scopedHeaders,
     };
   }
+
+  private getSecondaryAuthHeaders(request: ScopeableRequest): Headers {
+    const headerSource = isRealRequest(request)
+      ? this.authHeaders?.get(request) ?? {}
+      : request.headers;
+    const authorizationHeader = Object.entries(headerSource).find(([key, _]) => {
+      return key.toLowerCase() === AUTHORIZATION_HEADER;
+    });
+    const scopedHeaders = authorizationHeader
+      ? { [ES_SECONDARY_AUTH_HEADER]: authorizationHeader[1] }
+      : {};
+
+    return {
+      ...getDefaultHeaders(this.kibanaVersion),
+      ...this.config.customHeaders,
+      ...scopedHeaders,
+    };
```

Comment:
@azasypkin added you a reviewer to double check this.

This is the approach I took after our slack and GH discussion. For secondary-auth scoped clients, we're using the exact same credentials, configuration, and headers as the internal client. The only exception is the addition of the `es-secondary-authorization` header, with the value resolved from the request's `authorization` header.